packages/cli/src/commands/start.ts

```diff
@@ -246,36 +229,57 @@ export class Start extends BaseCommand {
 			config.set(setting.key, jsonParse(setting.value, { fallbackValue: setting.value }));
 		});
 
-		config.set('nodes.packagesMissing', '');
-		if (missingPackages.size) {
-			LoggerProxy.error(
-				'n8n detected that some packages are missing. For more information, visit https://docs.n8n.io/integrations/community-nodes/troubleshooting/',
-			);
+		const areCommunityPackagesEnabled = config.getEnv('nodes.communityPackages.enabled');
+
+		if (areCommunityPackagesEnabled) {
+			const installedPackages = await getAllInstalledPackages();
+			const missingPackages = new Set<{
+				packageName: string;
+				version: string;
+			}>();
+			installedPackages.forEach((installedPackage) => {
+				installedPackage.installedNodes.forEach((installedNode) => {
+					if (!this.loadNodesAndCredentials.known.nodes[installedNode.type]) {
+						// Leave the list ready for installing in case we need.
+						missingPackages.add({
+							packageName: installedPackage.packageName,
+							version: installedPackage.installedVersion,
+						});
+					}
+				});
+			});
 
-			if (flags.reinstallMissingPackages || process.env.N8N_REINSTALL_MISSING_PACKAGES) {
-				LoggerProxy.info('Attempting to reinstall missing packages', { missingPackages });
-				try {
-					// Optimistic approach - stop if any installation fails
-					// eslint-disable-next-line no-restricted-syntax
-					for (const missingPackage of missingPackages) {
-						await this.loadNodesAndCredentials.installNpmModule(
-							missingPackage.packageName,
-							missingPackage.version,
-						);
-						missingPackages.delete(missingPackage);
+			config.set('nodes.packagesMissing', '');
```

Comment:
Is there a chance that something breaks since this is only defined inside the block?
Previously, this would always be set, so in other parts of the code we could rely that this variable would always exist as an empty string.

Reply:
`config.get('nodes.packagesMissing')` is called from:

- `hasPackageLoaded()` at `POST /nodes`
- `removePackageFromMissingList()` at `POST /nodes`
- `getInstalledPackages()` at `GET /nodes`

And this nodes controller is conditional on the feature being enabled:

```ts
if (config.getEnv('nodes.communityPackages.enabled')) {
	controllers.push(
		new NodesController(config, this.loadNodesAndCredentials, this.push, internalHooks),
	);
}
```

So in theory it's fine. But if we want to make absolutely sure, I can also extract this line out of the block.